### PR TITLE
Vue i18n

### PR DESCRIPTION
**Pull request by ylavoie**
This PR adds internationalization to the Vue components.
- The current locale data is converted to JSON format and duplicated in Vue;
- Language data is generated and loaded on demand, defaulting to english
- A small language selected is provided to demonstrate functionality,

### CODE REVIEW

**Review by ehuelsmann — CHANGES_REQUESTED** [review]
There's at least one component missing: I tried to compile the sources (originally thought it would compile using `devdojo`), but can't test the branch.

File: UI/src/locales/de.json [hunk 1]
```diff
@@ -0,0 +1 @@
...nicht in der Kartei!"],"[_1]: Vendor not on file!":[null,"[_1]: Lieferant nicht in der Kartei!"],"action not defined!":[null,"Aktion nicht definiert!"],"balance":[null,"Balance"],"city":[null,"Stadt"],"csv":[null,""],"days":[null,"Tage"],"debits":[null,"Belastungen"],"default":[null,""],"delete":[null,"lÃ¶schen"],"depreciation":[null,""],"disposal":[null,""],"done":[null,"Erledigt"],"e":[null,""],"ea":[null,"ea"],"failed":[null,"fehlgeschlagen"],"from_date":[null,""],"html":[null,""],"import":[null,"importieren"],"in months":[null,"in Monaten"],"in use":[null,""],"in years":[null,"in Jahren"],"is bigger than the amount available":[null,"ist grÃ¶sser als der verfÃ¼gbare Betrag"],"is lesser than 0":[null,"ist kleiner als 0"],"is lesser than the amount to be paid":[null,"ist weniger als der zu zahlende Betrag"],"is null":[null,"ist null"],"label":[null,""],"partial disposal":[null,"teilweisige Entsorgung"],"permalink":[null,"Permalink"],"production":[null,"Produktion"],"sent":[null,"gesendet"],"tex":[null,""],"time":[null,"Zeit"],"time(s)":[null,"Zeit(en)"],"to_date":[null,""],"unexpected error!":[null,"unerwarteter Fehler!"],"use of an overpayment":[null,"Nutzen einer Ãœberzahlung"]}
```
Comment (ehuelsmann) [inline UI/src/locales/de.json:1]: in `{"30":[null,"30"],...}` what is the `null` about?

Comment (ylavoie) [reply UI/src/locales/de.json]: Leftover from full json format. Converted to standard.

File: webpack.config.js [hunk 2]
```diff
@@ -175,7 +189,55 @@ if (TARGET !== 'readme') {
         loaderConfig: require("./UI/js-src/lsmb/webpack.loaderConfig.js"),
         environment: { dojoRoot: "UI/js" }, // used at run time for non-packed resources (e.g. blank.gif)
         buildEnvironment: { dojoRoot: "node_modules" }, // used at build time
-        locales: ["en"],
+        locales: [
```
Comment (ehuelsmann) [inline webpack.config.js]: This is the current list of supported languages (should that be called locales?); shouldn't we make that dynamic?

File: UI/src/i18n.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    "ru",
+    "sv",
+    "tr",
+    "uk",
+    "zh_CN",
+    "zh_TW"
+];
+
+export function setI18nLanguage(i18n, locale) {
+    if (i18n.mode === "legacy") {
```
Comment (ehuelsmann) [inline UI/src/i18n.js:58]: As we only run on Vue3, does it make sense to support "legacy" mode?

File: UI/src/i18n.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        i18n.global.locale.value = locale;
+    }
+    /**
+     * NOTE:
+     * If you need to specify the language setting for headers, such as the `fetch` API, set it here.
+     * The following is an example for axios.
+     *
+     * axios.defaults.headers.common['Accept-Language'] = locale
+     */
+    document.querySelector("html").setAttribute("lang", locale);
```
Comment (ehuelsmann) [inline UI/src/i18n.js:72]: Our documents don't have the "lang" tag on the HTML attribute, which makes this line moot, unless we intend to start using the "lang" attribute?

File: UI/src/i18n.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+export function setupI18n(options = { locale: "en" }) {
+    const i18n = createI18n(options);
+    setI18nLanguage(i18n, options.locale);
```
Comment (ehuelsmann) [inline UI/src/i18n.js:77]: Why do you need this line, if the i18n object was just created using *exactly* those settings?

File: UI/src/i18n.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+export async function loadLocaleMessages(i18n, locale) {
+    // load locale messages with dynamic import
+    const messages = await import(
+        /* webpackChunkName: "locale-[request]" */ `./locales/${locale}.json`
+    );
+
+    // set locale and locale message
+    i18n.global.setLocaleMessage(locale, messages.default);
+
+    return nextTick();
```
Comment (ehuelsmann) [inline UI/src/i18n.js:90]: Why are you returning `nextTick()`? should `loadLocaleMessages()` return anything at all?

File: UI/src/locales/ar_EG.json [hunk 7]
```diff
@@ -0,0 +1,1817 @@
+{
+    "30": "30",
```
Comment (ehuelsmann) [inline UI/src/locales/ar_EG.json:2]: You converted the entire translation lexicon, but we have only a very small amount of translation on the client side as of yet. Genuine question: should we only send the bits for which we have actual translations? (Obviously, we can only present the english version for anything we didn't translate yet, so maybe there's no reason to even *send* it? also, for any keys that aren't being used on the client for translation today, maybe we need not send those either? That'd be more efficient, but I have no idea about the downsides...)

File: webpack.config.js (hunk 2 above)
Comment (ylavoie) [reply webpack.config.js]: Fixed

File: UI/src/i18n.js (hunk 3 above)
Comment (ylavoie) [reply UI/src/i18n.js]: No, it doesn't. Fixed

**Comment by ylavoie** [comment]
This PR badly needs cleaning and be splitted to address different concerns. Closing.

**Pull request closed without merging**